pex/compatibility.py

```diff
@@ -8,28 +8,35 @@
 
 import os
 from abc import ABCMeta
-from io import StringIO
 from sys import version_info as sys_version_info
 
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+
 try:
-    # Python 2.x
-    from ConfigParser import ConfigParser
-except ImportError:
     # Python 3.x
-    from configparser import ConfigParser
+    from configparser import ConfigParser as ConfigParser  # type: ignore[import]
+except ImportError:
+    # Python 2.x
+    from ConfigParser import ConfigParser as ConfigParser  # type: ignore[no-redef]
```

Comment:
Note that this reordering was necessary, as the alternative was impossible to fix by adding ignores.

Reply:
Well, this could probably be switched back now that we've unified the `mypy.ini` for both versions.

Reply:
Didn't realize it was reordered. I'll try to make the diff smaller.